### DISCUSSION TITLE
re-process all window-rules

### DISCUSSION THREAD

**Discussion by permezel** [YaLTeR/niri#732] [Ideas]
Would be nice to be able to re-process all window rules.
I keep wrecking my mapping of windows to workspaces (eventually I will learn not to I suppose) but it would be nice to just have all the window-rules "open-on-workspace" be re-applied, say by writing the config file.
I would be happy with a "on-reconfig" match, and put what I needed to be done in there.

**Comment by YaLTeR** [comment]
Well, window rules are always live-reloaded and live-applied. It's just that the `open-on` rules, as the name suggests, only apply at the time when the window *opens*.

I understand your problem, so maybe indeed there should be some command that makes it re-apply all the open-on rules. What could it be called? `reapply-open-on-rules`? Should it be a debug command maybe? Should it apply to open-on-output and to named workspaces, or should there be a window-specific command?

**Reply by permezel** [reply]
Perhaps certain `open-on` could be flagged with `--relocate=true` and  there be a `relocate-wayward-windows` command?
That way, only the ones I desire a strict workspace mappings would be relocated.  Not sure "relocated' is the correct verb.  "rehomed"?

Alternatively, `open-on` could accept a `--lockdown=true` and then the workspace movement commands would fail.

I would not make it a debug command.
It should work for both named and numbered work spaces.
